nntrainer/tensor/tensor_v2.cpp

```diff
@@ -90,6 +90,23 @@ TensorV2::TensorV2(
 }
 #endif
 
+bool TensorV2::operator==(const TensorV2 &rhs) const {
+  /// compares tensor information
+  if (*itensor == *rhs.itensor) {
+    /// compares tensor data
+    if (getDataType() == Tdatatype::FP32) {
+      return *dynamic_cast<FloatTensor *>(itensor) ==
+             *dynamic_cast<FloatTensor *>(rhs.itensor);
+    } else if (getDataType() == Tdatatype::FP16) {
+#ifdef ENABLE_FP16
+      return *dynamic_cast<HalfTensor *>(itensor) ==
+             *dynamic_cast<HalfTensor *>(rhs.itensor);
+#endif
+    }
+  }
+  return false;
+}
+
```

Comment:
How about using switch-case statement ? 
```.cpp
bool TensorV2::operator==(const TensorV2& rhs) const {  
    // Compare tensor information  
    if (*itensor == *rhs.itensor) {  
        // Compare tensor data based on data type  
        switch (getDataType()) {  
            case Tdatatype::FP32:  
                return *dynamic_cast<FloatTensor*>(itensor) ==   
                       *dynamic_cast<FloatTensor*>(rhs.itensor);  
            case Tdatatype::FP16:  
#ifdef ENABLE_FP16  
                return *dynamic_cast<HalfTensor*>(itensor) ==   
                       *dynamic_cast<HalfTensor*>(rhs.itensor);  
#else  
                break;  
#endif  
            default:  
                break;  
        }  
    }  
    return false;  
}  
```

The "Tensor == operation" can be used frequently and I believe that a switch case statement is more efficient when it comes to compiler optimization. Furthermore, I think that a switch case statement may be more appropriate semantically.

Reply:
In general cases, if statements can be faster if there are two conditions.
There isn't much difference in speed on recent computers, so wouldn't it be better to have code that is easy to read?